chapters/101/index.md

```diff
@@ -1,167 +1,113 @@
 
-In Jina, **a Document is anything that you want to search for**: a text document, a short tweet, a code snippet an image, a video/audio clip, GPS traces of a day. A Document is also the input query when searching.
+A good example is the text we type into a search engine like Google. Documents can also be video clips, genetic sequences, MP3 files, scientific papers, funny GIFs, or many other things.
 
-**A Chunk is a small semantic unit of a Document.** It could be a sentence, a 64x64 image patch, a 3 second video shot, a pair of coordinate and address. 
+You can think of a Document as a chocolate bar: Different Documents have different formats or ingredients, and you can [break them into smaller chunks](#segmenter).
 
-In Jina, a Document is like chocolate bar. Not only because it comes with different kinds and ingredients, but also you can break it into chunks in the way you like. Eventually, what you buy and store are the chocolate bars, and what you eat and digest are the chunks. You don’t want to swallow the whole bar, you also don’t want to cut it into powder; either way you will lose the flavor (i.e. the semantic).
 
+## Flow
 
-## YAML config
-
-<span style="display: inline-block; width: 30%;">
-
-![Jina 101 YAML, Copyright by Jina AI Limited](./img/ILLUS2.png)
+![Flow Cartoon](images/flow.svg)
 
-</span>
+Flows manage the two main search workflows: indexing data (indexing Flow) and then querying it (querying Flow).
 
-**A YAML config is widely used in Jina to describe the properties of an object.** It offers customization, allowing users to change the behavior of an object without touching its code. Jina can build a very complicated object directly from a simple YAML config, and save an object into a YAML config.
+A Flow is made up of several sub-tasks, and it manages the states and context of these sub-tasks. The input and output data of Flows are Documents.
 
 
 ## Executor
 
-<span style="display: inline-block; width: 30%;">
+Executors perform each task in an indexing or querying Flow. Jina offers [many Executors](https://docs.jina.ai/chapters/all_exec.html), which can be divided as follows:
 
-![Jina AI Executor, Copyright by Jina AI Limited](./img/ILLUS3.png)
 
-</span>
+### Crafter
 
+![Crafter Cartoon](images/crafter.svg)
 
-**Executor represents an algorithmic unit in Jina.** Algorithms such as encoding images into vectors, storing vectors on the disk, ranking results, can all be formulated as Executors. Executor provides useful interfaces, allowing AI developers and engineers to really focus on the algorithm. Features such as persistency, scheduling, chaining, grouping and parallelization come out of the box.
+Crafters pre-process input Documents, for example, resizing images or converting text to lower case. A Crafter often comes before the [Encoder](images/#encoder) but it’s not always required.
```

Comment:
technically it can also do post processing

Reply:
@JoanFM do you have some examples of when?

Reply:
well it is agnostic to when it is done, I just inform but it may make sense to communicate them as preprocessors.

I just like to comment to put it on the table

Reply:
@JoanFM I see. 

So, I understand it this way - 
Users have their freedom to arrange Executors the way they want, so technically it's possible do post processing. But in the search workflows, a Crafter is **often** doing preprocessing.

Reply:
Let's keep it simple for 101. We can talk about Executors in depth in another doc